Source/WebKit/GPUProcess/media/RemoteAudioVideoRendererProxyManager.cpp

```diff
@@ -0,0 +1,576 @@
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setMuted(muted);
+}
+
+void RemoteAudioVideoRendererProxyManager::setPreservesPitchAndCorrectionAlgorithm(RemoteAudioVideoRendererIdentifier identifier, bool preservesPitch, std::optional<WebCore::MediaPlayerPitchCorrectionAlgorithm> algorithm)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setPreservesPitchAndCorrectionAlgorithm(preservesPitch, algorithm);
+}
+
+#if HAVE(AUDIO_OUTPUT_DEVICE_UNIQUE_ID)
+void RemoteAudioVideoRendererProxyManager::setOutputDeviceId(RemoteAudioVideoRendererIdentifier identifier, const String& deviceId)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setOutputDeviceId(deviceId);
+}
+#endif
+
+void RemoteAudioVideoRendererProxyManager::setIsVisible(RemoteAudioVideoRendererIdentifier identifier, bool visible)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setIsVisible(visible);
+}
+
+void RemoteAudioVideoRendererProxyManager::setPresentationSize(RemoteAudioVideoRendererIdentifier identifier, const WebCore::IntSize& size)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setPresentationSize(size);
+}
+
+void RemoteAudioVideoRendererProxyManager::setShouldMaintainAspectRatio(RemoteAudioVideoRendererIdentifier identifier, bool maintain)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setShouldMaintainAspectRatio(maintain);
+}
+
+void RemoteAudioVideoRendererProxyManager::acceleratedRenderingStateChanged(RemoteAudioVideoRendererIdentifier identifier, bool renderingIsAccelerated)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->acceleratedRenderingStateChanged(renderingIsAccelerated);
+}
+
+void RemoteAudioVideoRendererProxyManager::contentBoxRectChanged(RemoteAudioVideoRendererIdentifier identifier, const WebCore::LayoutRect& rect)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->contentBoxRectChanged(rect);
+}
+
+void RemoteAudioVideoRendererProxyManager::notifyWhenHasAvailableVideoFrame(WebKit::RemoteAudioVideoRendererIdentifier identifier, bool notify)
+{
+    if (RefPtr renderer = rendererFor(identifier)) {
+        if (notify) {
+            renderer->notifyWhenHasAvailableVideoFrame([weakThis = WeakPtr { *this }, identifier](auto presentationTime, auto clockTime) {
+                if (RefPtr protectedThis = weakThis.get()) {
+                    if (!protectedThis->m_renderers.contains(identifier))
+                        return;
+                    protectedThis->m_gpuConnectionToWebProcess.get()->connection().send(Messages::AudioVideoRendererRemoteMessageReceiver::HasAvailableVideoFrame(presentationTime, clockTime, protectedThis->stateFor(identifier)), identifier);
+                }
+            });
+            return;
+        }
+        renderer->notifyWhenHasAvailableVideoFrame({ });
+    }
```

Comment:
I would prefer early returns over deeply nested if statements. Something like:

```suggestion
    RefPtr renderer = rendererFor(identifier);
    if (!renderer)
        return;
    if (!notify) {
        renderer->notifyWhenHasAvailableVideoFrame({ });
        return;
    }

    renderer->notifyWhenHasAvailableVideoFrame([weakThis = WeakPtr { *this }, identifier](auto presentationTime, auto clockTime) {
        if (RefPtr protectedThis = weakThis.get(); protectedThis && protectedThis->m_renderers.contains(identifier))
            protectedThis->m_gpuConnectionToWebProcess.get()->connection().send(Messages::AudioVideoRendererRemoteMessageReceiver::HasAvailableVideoFrame(presentationTime, clockTime, protectedThis->stateFor(identifier)), identifier);
    });
```

Reply:
I re-wrote them all to always exit early, except if it's a one-liner